### PR TITLE
Clean up warnings in `saw-remote-api`

### PR DESCRIPTION
**Pull request by atomb**

### CODE REVIEW

File: saw-remote-api/src/SAWServer.hs [hunk 1]
```diff
@@ -187,6 +186,8 @@ initialState readFile =
 #else
                 , roProxy = AIGProxy AIG.basicProxy
 #endif
+                , roInitWorkDir = cwd
+                , roBasicSS = undefined
```
Comment (brianhuffman) [inline saw-remote-api/src/SAWServer.hs:190]: We probably shouldn't be using `undefined` anywhere. Do you have a plan for what should go here instead?

File: saw-remote-api/src/SAWServer.hs [hunk 2]
```diff
@@ -202,6 +203,8 @@ initialState readFile =
                 , rwCrucibleAssertThenAssume = False
                 , rwLaxArith = False
                 , rwWhat4HashConsing = False
+                , rwProofs = undefined
+                , rwPreservedRegs = undefined
```
Comment (brianhuffman) [inline saw-remote-api/src/SAWServer.hs:207]: Same comment about `undefined` here.

File: saw-remote-api/src/SAWServer.hs (hunk 1 above)
Comment (pnwamk) [reply saw-remote-api/src/SAWServer.hs:190]: FYI, for this line this has been working for me in my recent tinkering: https://github.com/GaloisInc/saw-script/commit/d4a6f8b5a855c647563dce25c7ab82787f90555f

Comment (atomb) [reply saw-remote-api/src/SAWServer.hs:190]: Oh, oops! I need to fill that in.

File: saw-remote-api/src/SAWServer.hs (hunk 2 above)
Comment (atomb) [reply saw-remote-api/src/SAWServer.hs:207]: Same answer. :)

**Review by brianhuffman — APPROVED** [review]

**Pull request merged**